homeassistant/components/kostal_plenticore/config_flow.py

```diff
@@ -32,9 +32,17 @@ async def test_connection(hass: HomeAssistant, data) -> str:
     session = async_get_clientsession(hass)
     async with ApiClient(session, data["host"]) as client:
         await client.login(data["password"])
-        values = await client.get_setting_values("scb:network", "Hostname")
 
-    return values["scb:network"]["Hostname"]
+        # get name of the hostname setting id
+        all_settings = await client.get_settings()
+        for entry in all_settings["scb:network"]:
+            if "Hostname" in entry.id:
```

Comment:
I am not happy with this "heuristic". I would prefer an explicit check against the known ids "Network:Hostname" and "Hostname".

Maybe something like this:

```python
KNOWN_HOSTNAME_IDS = ("Network:Hostname", "Hostname")

def get_hostname_id(client: ApiClient) -> str:
    all_settings = await client.get_settings()
    for entry in all_settings["scb:network"]:
        if entry.id in KNOWN_HOSTNAME_IDS:
            return entry.id
    throw Some Exception
```